### ISSUE TITLE
Account sign up issues

### ISSUE THREAD

**Issue by starryeyez024** [htothenan1/min-waste#5]
1. When someone tries to sign up with an email address which is already in use, it shows a non-helpful error instead of a message about just logging in with that email address

<img width="783" alt="image" src="https://github.com/htothenan1/min-waste/assets/456863/86f2289d-a12c-46e4-a9ee-51f79b629bea">

2. There is no way to reset your password if you forget

**Comment by htothenan1** [comment]
1) You don't think "Oops, something went wrong!" is helpful enough? lol I think this is one of the first things I did, so I have no idea why I was so unhelpful with those words. The only error you can have if you fill out all three fields, is if the email already exists, so I don't think there's any reason to not just change the text.

2) Yeah, would love to hear your preferred option here, seeing that I am not requiring a real email address to sign in. Maybe we just need to require a real email, so we can have a proper forget password flow?

**Comment by starryeyez024** [comment]
Yep, i mean idk the level of effort but the ability to reset a password would be a good-to-have feature!

**Comment by htothenan1** [comment]
for sure. My thinking back then was to allow fake email registers to make it easier for anyone to kind of test it out, without putting real info. 

I suppose if we wanted to have a password reset feature, we would need to have some sort of confirmation email at registration time for the user. That way, if they forget password, we could just send a reset email. 

Are there other ways that you know of?

**Comment by starryeyez024** [comment]
I can see your logic, though I don't know that people *have* to confirm their email first in order to have a password reset later. If someone inputs a junk email and then later forgets their password, then they miss out on the chance to reset it (and they would just have to create another account).

The one fail-safe is you would want to ensure that the email address already exists as a user in your database before sending a pw reset email, just to avoid your servers being overtaken by some bot.

**Comment by htothenan1** [comment]
that's an excellent point. We could just send the reset email, even if its not real. Wait. There is the edge case that someone uses someone elses real email... nothing to worry about?

**Comment by starryeyez024** [comment]
> There is the edge case that someone uses someone elses real email... nothing to worry about?
Right! If i get a password reset email that I didn't initiate, I can either ignore it, or change my password to something more secure knowing someone is trying to hack my account. :)
